src/main/kotlin/view/MineSweeperController.kt

```diff
@@ -1,16 +1,36 @@
 package view
 
-import domain.map.MineMapFactory
+import domain.game.MineSweeperGameFactory
+import domain.game.OpenResult
 
 class MineSweeperController(
     private val inputView: MineSweeperInputView,
     private val resultView: MineSweeperResultView,
-    private val mineMapFactory: MineMapFactory,
+    private val mineSweeperGameFactory: MineSweeperGameFactory,
 ) {
 
     fun start() {
         val mineSweeperInitProperty = inputView.readInitProperty()
-        val mineMap = mineMapFactory.create(mineSweeperInitProperty)
-        resultView.display(mineMap)
+        resultView.displayStartMineSweeperGameMessage()
+
+        val mineSweeperGame = mineSweeperGameFactory.create(mineSweeperInitProperty)
+
+        while (true) {
+            val coordinate = inputView.readOpenCoordinate()
+            when (val openResult = mineSweeperGame.open(coordinate)) {
+                is OpenResult.MineOpened -> {
+                    resultView.displayLoseGameMessage()
+                    return
+                }
+                is OpenResult.GroundOpened -> {
+                    resultView.display(openResult.mapCapture)
+                    continue
+                }
+                is OpenResult.AllMineFound -> {
+                    resultView.displayWinGameMessage()
+                    return
+                }
+            }
+        }
```

Comment:
game 객체가 지뢰찾기 게임을 수행하도록 구현해보면 좋을 것 같아요. 😃 (지금의 경우에는 `MineMap`과 `MineSweeperGame` 객체의 차이를 잘 모르겠어요. 🤔

mineSweeperGame 객체에게 게임이 진행중인 상태인지 메시지를 보내서 알아보도록 변경하고, OpenResult를 enum class로 바꾼다는 가정하에 아래와 같이 변경해보면 어떨까요? 

(지금의 상태에서 코드를 크게 변경하지 않으면서 객체에게 메시지를 보내는 방식으로 변경해보려니 쉽지가 않네요. 😅)

```suggestion
        while (mineSweeperGame.isPlaying()) {
            val coordinate = inputView.readOpenCoordinate()
            val openResult = mineSweeperGame.open(coordinate)
            when(openResult) {
                OpenResult.GROUND_OPEND -> resultView.display(mineSweeperGame)
                OpenResult.MINE_OPEND -> resultView. displayLoseGameMessage()
                OpenResult.ALL_GROUND_OPEND -> resultView. displayWinGameMessage()
            }
        }
```

Reply:
결과값에 게임 진행 가능 여부 값을 넣는 방법으로 개선했습니다~!
https://github.com/next-step/kotlin-minesweeper/pull/298/commits/f632ef1f83373a44bd42a2f66d721703baa818e8